hyperspy/drawing/utils.py

```diff
@@ -500,7 +507,15 @@ def plot_images(images,
 
     # Get default colormap from pyplot:
     if cmap is None:
-        cmap = plt.get_cmap()
+        cmap = plt.get_cmap().name
+    elif isinstance(cmap, mpl.colors.Colormap):
+        cmap = cmap.name
+    if centre_colormap == "auto":
+        if cmap in MPL_DIVERGING_COLORMAPS:
+            centre_colormap = True
+        else:
+            centre_colormap = False
+        print centre_colormap
```

Comment:
We could turn the code into a function but, although this is a good
idea in general, in this case I feel that it can obscure the code
because it changes the value of two local variables, `cmap` and
`centre_colormap`.
On Mon, 2015-08-10 at 09:41 -0700, Tomas Ostasevicius wrote:
> In hyperspy/drawing/utils.py:
> > @@ -500,7 +507,15 @@ def plot_images(images,
> >  
> >      # Get default colormap from pyplot:
> >      if cmap is None:
> > -        cmap = plt.get_cmap()
> > +        cmap = plt.get_cmap().name
> > +    elif isinstance(cmap, mpl.colors.Colormap):
> > +        cmap = cmap.name
> > +    if centre_colormap == "auto":
> > +        if cmap in MPL_DIVERGING_COLORMAPS:
> > +            centre_colormap = True
> > +        else:
> > +            centre_colormap = False
> > +        print centre_colormap
> This whole dance with "auto" gets done twice in the code... Any way
> to combine the two?
> Also print here as well
> —
> Reply to this email directly or view it on GitHub.